### ISSUE TITLE
module 'types' has no attribute 'UnionType'

### ISSUE THREAD

**Issue by Tiffinini** [Pycord-Development/pycord#1239]
### Summary

A conditional check in enums.py is causing Python versions <3.10 to crash.

### Reproduction Steps

1. Define a slash command with an option.
2. Run.

### Minimal Reproducible Code
In my case, this error occurs when defining any slash command using the Option parameter:
```python
    @slash_command(guild_ids=globals.COMMAND_GUILD_ID_REGISTER_LIST)
    async def note(self, ctx, note: Option(str, "Add a note.")):
```

### Expected Results

The code to run, as it did under b5.

### Actual Results

Crash with:

module 'types' has no attribute 'UnionType'

### Intents

All

### System Information

- Python v3.9.2-final
- py-cord v2.0.0-beta
    - py-cord pkg_resources: v2.0.0b6
- aiohttp v3.8.1
- system info: Windows 8.1 6.3.9600

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

### Additional Context

Line 672 of enums.py makes a check to see if datatype is of type: types.UnionType. UnionType is not part of the types built-in for Python versions below 3.10, causing: module 'types' has no attribute 'UnionType'. 

Judging by the conditional check to begin with, I assume the intention is not to make 3.10+ a requirement to use this library.

**Comment by EmreTech** [comment]
Well, you beat me to making this issue. Congrats

**Comment by Lulalaby** [comment]
Have fun emre

**Comment by EmreTech** [comment]
Fixed by #1240
